packages/gator-permissions-snap/src/core/permissionHandler.ts

```diff
@@ -287,8 +263,9 @@ export class PermissionHandler<
         await updateContext({ updatedContext: currentContext });
       };
 
-      // fetchAccountBalanceCallCounter is used to cancel any previously executed instances of this function.
-      let fetchAccountBalanceCallCounter = 0;
+      const balanceOperation = createCancellableOperation<TContext>();
```

Comment:
Related to the comments in the `cancellableOperation` file - how would you feel about defining the execute and callback functions when you instantiate the cancellable operation?

Then when you want to execute, you just pass the `args`.

```
const myOperation = createCancellableOperation<TArgs>({ execute, callback });

await myOperation(args);
```